internal/operators/lvm/common.go

```diff
@@ -0,0 +1,67 @@
+package lvm
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+)
+
+type baseHardware struct {
+	CPUCores int64 `json:"cpu_cores"`
+	RAMMib   int64 `json:"ram_mib"`
+}
+
+type hardwareRequirement struct {
+	Version string       `json:"version,omitempty"`
+	Master  baseHardware `json:"master"`
+	Worker  baseHardware `json:"worker"`
+	Sno     baseHardware `json:"sno"`
+}
+
+const (
+	envRequirementsKey string = "HW_VALIDATOR_REQUIREMENTS"
+	urlRequirement     string = "https://raw.githubusercontent.com/openshift/assisted-service/master/data/default_hw_requirements.json"
+
+	openshiftVersion4_10 string = "4.10.0"
+	openshiftVersion4_11 string = "4.11.0"
+	openshiftVersion4_13 string = "4.13.0"
+	openshiftVersion4_15 string = "4.15.0"
+)
+
+//TODO: move to common and use it for all operators
+
+type pageDownloader func(s string) (*http.Response, error)
+
+func downloadFromUrl(pageDownloadFunc pageDownloader) (*http.Response, error) {
+	return pageDownloadFunc(urlRequirement)
+}
+
+func GetPage(url string) (*http.Response, error) {
+	return http.Get(url)
+}
+
+func GetBaseRequiemnts(pageDownloadFunc pageDownloader) (hardwareRequirement, error) {
+	var baseHardwareRequirement = make([]hardwareRequirement, 1)
+
+	baseRequirement := os.Getenv(envRequirementsKey)
+	if baseRequirement != "" {
+		err := json.Unmarshal([]byte(baseRequirement), &baseHardwareRequirement)
+		if err == nil {
+			return baseHardwareRequirement[0], nil
+		}
+	}
+	resp, err := downloadFromUrl(pageDownloadFunc)
+	if err != nil {
+		return baseHardwareRequirement[0], fmt.Errorf("Couldn't download requirement file")
+	}
```

Comment:
I don't think we should download files from github at runtime or for the tests, download can fail depending on github status, and because of rate limiting policies. Also, it can break for unrelated reasons if someone moves/rename the file.

I think we should enforce `envRequirementsKey` to be set when we start the service.